vulnerabilities/importers/apache_kafka.py

```diff
@@ -7,120 +7,177 @@
 
-    def updated_advisories(self):
-        advisory_page = self.fetch_advisory_page()
-        self.set_api()
         parsed_data = self.to_advisory(advisory_page)
-        return self.batch_advisories(parsed_data)
+        return parsed_data
 
     def to_advisory(self, advisory_page):
         advisories = []
+
         advisory_page = BeautifulSoup(advisory_page, features="lxml")
         cve_section_beginnings = advisory_page.find_all("h2")
         for cve_section_beginning in cve_section_beginnings:
-            cve_id = cve_section_beginning.text.split("\n")[0]
+            # This sometimes includes text that follows the CVE on the same line -- sometimes there is a carriage return, sometimes there is not
+            # cve_id = cve_section_beginning.text.split("\n")[0]
+            # This is superior, gets only the cve id and no following text.
+            cve_id = cve_section_beginning.get("id")
+
             cve_description_paragraph = cve_section_beginning.find_next_sibling("p")
+
+            stripped_cve_description_paragraph = str(cve_description_paragraph.get_text())
+            stripped_cve_description_paragraph = stripped_cve_description_paragraph.replace(
+                "\n", ""
+            )
+            stripped_cve_description_paragraph = " ".join(
+                stripped_cve_description_paragraph.split()
+            )
+
             cve_data_table = cve_section_beginning.find_next_sibling("table")
             cve_data_table_rows = cve_data_table.find_all("tr")
             affected_versions_row = cve_data_table_rows[0]
             fixed_versions_row = cve_data_table_rows[1]
-            affected_version_ranges = to_version_ranges(
-                affected_versions_row.find_all("td")[1].text
-            )
-            fixed_version_ranges = to_version_ranges(fixed_versions_row.find_all("td")[1].text)
-
-            fixed_packages = [
-                PackageURL(type="apache", name="kafka", version=version)
-                for version in self.version_api.get("apache/kafka").valid_versions
-                if any(
-                    [
-                        MavenVersion(version) in version_range
-                        for version_range in fixed_version_ranges
-                    ]
-                )
-            ]
 
-            affected_packages = [
-                PackageURL(type="apache", name="kafka", version=version)
-                for version in self.version_api.get("apache/kafka").valid_versions
-                if any(
-                    [
-                        MavenVersion(version) in version_range
-                        for version_range in affected_version_ranges
-                    ]
-                )
+            affected_versions_string = affected_versions_row.find_all("td")[1].text
+            fixed_versions_string = fixed_versions_row.find_all("td")[1].text
+
+            # Remove leading white space after initial comma
+            affected_versions_string_split_SPLIT = [
```

Comment:
This is a very variable name.... Consider this code instead:
```
            affected_versions = [v.strip() for v in affected_versions.split(",")]
            affected_versions = [v for v in affected_versions if v]
```

Reply:
@pombredanne After making your suggested changes I get an error re my hard-coded mapping: 

```
TypeError: unhashable type: 'list'
```

Replacing the awkwardly-named

```
            affected_versions_string_split_SPLIT = [
                substring.strip()
                for substring in affected_versions.split(",")
                if not substring.isspace()
            ]
```

with 

```
            affected_versions = [v.strip() for v in affected_versions.split(",")]
            affected_versions = [v for v in affected_versions if v]
```

has converted the previously-defined `affected_versions = affected_versions_row.find_all("td")[1].text` to a list, which won't work as a mapping/dict key.  There's probably a simple fix for this, giving it some thought now....

Reply:
@pombredanne This is where the conversion of `affected_versions` to a list throws an error.  It also applies to your subsequent comment re `fixed_versions_string_split_SPLIT`.  

```
            # This throws a KeyError if the opening h2 tag `id` data changes or is not in the
            # hard-coded affected_version_range_mapping dictionary.
            if affected_version_range_mapping[cve_id]["action"] == "include":

                # These 2 variables (not used elsewhere) trigger the KeyError for changed/missing data.
                check_affected_versions_key = affected_version_range_mapping[cve_id][
                    affected_versions
                ]
                check_fixed_versions_key = affected_version_range_mapping[cve_id][
                    fixed_versions_string
                ]
```

I don't see how to modify the excerpted code to deal with your conversion of `affected_versions` to a list.

Reply:
Hmm, by simply changing the variable name ever so slightly to avoid using `affected_versions`, your approach works.  Am I missing something?  ;-)  This way we use your cleaner code AND avoid converting `affected_versions` itself to an unwanted list.

```
            affected_versions_clean = [v.strip() for v in affected_versions.split(",")]
            affected_versions_clean = [v for v in affected_versions if v]
```